uint32 disable_control
int32 velocity
